doc: error-handling.md: main case analysis for search

jrburke opened the pull request:
In src/doc/trpl/error-handling.md, in this section:

## Error handling with Box<Error>

It mentions three steps, to "convert this to proper error handling", the last one being:

3. Handle the error in main.

However, it is not shown. This pull request adds a code example showing how `main`'s call to `search` should use case analysis. I am still very much new to learning Rust, so this may not be idiomatic. Happy to make changes with guidance.

rust-highfive commented:
Thanks for the pull request, and welcome! The Rust team is excited to review your changes, and you should hear from @nikomatsakis (or someone else) soon.

If any changes to this PR are deemed necessary, please add them as extra commits. This ensures that the reviewer can see what has changed since they last reviewed the code. The way Github handles out-of-date commits, this should also make it reasonably obvious what issues have or haven't been addressed. Large or tricky changes may require several passes of review and changes.

Please see [the contribution instructions](https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md) for more information.

Manishearth commented:
r? @BurntSushi

BurntSushi commented on src/doc/trpl/error-handling.md:
```diff
@@ -1838,6 +1838,22 @@ impl<'a, 'b> From<&'b str> for Box<Error + Send + Sync + 'a>
+            println!("{}, {}: {:?}", pop.city, pop.country, pop.count);
+        }
+    }
+    Err(err) => panic!("{}", err)
```
Can you change this to `println!("{}", err)`? I don't think there's any reason to panic here, which would present an ugly error message to the user.

BurntSushi commented:
@jrburke Looks good, thank you! I had one small nit, but otherwise this looks good to go.

jrburke commented:
@BurntSushi I changed `panic` to `println`, then I rebased and squashed over latest master. I thought I read somewhere about keeping commits separate during review, but cannot find that now, and was unsure if squashing to one commit after review for bot merging was preferred.

Manishearth commented:
@bors r=BurntSushi rollup

Thanks!

bors commented:
:pushpin: Commit 0dd2c1c has been approved by `BurntSushi`

<!-- @bors r=BurntSushi 0dd2c1c07dd2483618790262a2a661fee34c992f -->

bors commented:
:hourglass: Testing commit 0dd2c1c with merge 9eb099d...

bors commented:
:sunny: Test successful - [auto-linux-32-nopt-t](http://buildbot.rust-lang.org/builders/auto-linux-32-nopt-t/builds/7037), [auto-linux-32-opt](http://buildbot.rust-lang.org/builders/auto-linux-32-opt/builds/7051), [auto-linux-64-debug-opt](http://buildbot.rust-lang.org/builders/auto-linux-64-debug-opt/builds/310), [auto-linux-64-nopt-t](http://buildbot.rust-lang.org/builders/auto-linux-64-nopt-t/builds/6989), [auto-linux-64-opt](http://buildbot.rust-lang.org/builders/auto-linux-64-opt/builds/7012), [auto-linux-64-x-android-t](http://buildbot.rust-lang.org/builders/auto-linux-64-x-android-t/builds/6997), [auto-linux-cross-opt](http://buildbot.rust-lang.org/builders/auto-linux-cross-opt/builds/228), [auto-linux-musl-64-opt](http://buildbot.rust-lang.org/builders/auto-linux-musl-64-opt/builds/2178), [auto-mac-32-opt](http://buildbot.rust-lang.org/builders/auto-mac-32-opt/builds/7066), [auto-mac-64-nopt-t](http://buildbot.rust-lang.org/builders/auto-mac-64-nopt-t/builds/7060), [auto-mac-64-opt](http://buildbot.rust-lang.org/builders/auto-mac-64-opt/builds/7027), [auto-win-gnu-32-nopt-t](http://buildbot.rust-lang.org/builders/auto-win-gnu-32-nopt-t/builds/2019), [auto-win-gnu-32-opt](http://buildbot.rust-lang.org/builders/auto-win-gnu-32-opt/builds/2022), [auto-win-gnu-64-nopt-t](http://buildbot.rust-lang.org/builders/auto-win-gnu-64-nopt-t/builds/2023), [auto-win-gnu-64-opt](http://buildbot.rust-lang.org/builders/auto-win-gnu-64-opt/builds/2018), [auto-win-msvc-32-opt](http://buildbot.rust-lang.org/builders/auto-win-msvc-32-opt/builds/1279), [auto-win-msvc-64-opt](http://buildbot.rust-lang.org/builders/auto-win-msvc-64-opt/builds/2005)

bors commented:
:sunny: Test successful - [auto-linux-32-nopt-t](http://buildbot.rust-lang.org/builders/auto-linux-32-nopt-t/builds/7037), [auto-linux-32-opt](http://buildbot.rust-lang.org/builders/auto-linux-32-opt/builds/7051), [auto-linux-64-debug-opt](http://buildbot.rust-lang.org/builders/auto-linux-64-debug-opt/builds/310), [auto-linux-64-nopt-t](http://buildbot.rust-lang.org/builders/auto-linux-64-nopt-t/builds/6989), [auto-linux-64-opt](http://buildbot.rust-lang.org/builders/auto-linux-64-opt/builds/7012), [auto-linux-64-x-android-t](http://buildbot.rust-lang.org/builders/auto-linux-64-x-android-t/builds/6997), [auto-linux-cross-opt](http://buildbot.rust-lang.org/builders/auto-linux-cross-opt/builds/228), [auto-linux-musl-64-opt](http://buildbot.rust-lang.org/builders/auto-linux-musl-64-opt/builds/2178), [auto-mac-32-opt](http://buildbot.rust-lang.org/builders/auto-mac-32-opt/builds/7066), [auto-mac-64-nopt-t](http://buildbot.rust-lang.org/builders/auto-mac-64-nopt-t/builds/7060), [auto-mac-64-opt](http://buildbot.rust-lang.org/builders/auto-mac-64-opt/builds/7027), [auto-win-gnu-32-nopt-t](http://buildbot.rust-lang.org/builders/auto-win-gnu-32-nopt-t/builds/2019), [auto-win-gnu-32-opt](http://buildbot.rust-lang.org/builders/auto-win-gnu-32-opt/builds/2022), [auto-win-gnu-64-nopt-t](http://buildbot.rust-lang.org/builders/auto-win-gnu-64-nopt-t/builds/2023), [auto-win-gnu-64-opt](http://buildbot.rust-lang.org/builders/auto-win-gnu-64-opt/builds/2018), [auto-win-msvc-32-opt](http://buildbot.rust-lang.org/builders/auto-win-msvc-32-opt/builds/1279), [auto-win-msvc-64-opt](http://buildbot.rust-lang.org/builders/auto-win-msvc-64-opt/builds/2005)
